app/models/blazer/check.rb

```diff
@@ -68,8 +68,13 @@ def update_state(result)
 
       # do not notify on creation, except when not passing
       if (state_was != "new" || state != "passing") && state != state_was && emails.present?
-        Blazer::CheckMailer.state_change(self, state, state_was, result.rows.size, message, result.columns, result.rows.first(10).as_json, result.column_types, check_type).deliver_now
-        Blazer::SlackNotifier.state_change(self, state, state_was, result.rows.size, message, check_type)
+        if emails.present?
+          Blazer::CheckMailer.state_change(self, state, state_was, result.rows.size, message, result.columns, result.rows.first(10).as_json, result.column_types, check_type).deliver_now
+        end
+
+        if slack_channels.present?
```

Comment:
Better to use `split_slack_channels.any?` for backwards compatibility, as older installs won't have the `slack_channels` column.